### ISSUE TITLE
Rule Request: Disallow `NSApp.activate(ignoringOtherApps: true)` in `applicationDidFinishLaunching`

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#2643]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Most macOS apps abuse this method. The problem is the `ignoringOtherApps: true` part. It means if you launch the app and then switch to a different app while it's launching, it will steal focus when it has finished launching. This is something that is hard to catch in development, but users gets annoyed by this behavior.

I propose reporting when `NSApp.activate(ignoringOtherApps: true)` is used inside `applicationDidFinishLaunching`. It's ok to use `ignoringOtherApps: true` in other situations. It's also ok to use `NSApp.activate(ignoringOtherApps: false)` inside `applicationDidFinishLaunching`.

Having this rule will hopefully make devs aware they're using the API incorrectly and fix their apps.

> 1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Because it's a widespread problem that affects real users: https://twitter.com/sindresorhus/status/1085002325511426051

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:

```swift
@NSApplicationMain
final class AppDelegate: NSObject, NSApplicationDelegate {
	func applicationDidFinishLaunching(_ notification: Notification) {
		NSApp.activate(ignoringOtherApps: true)
	}
}
```

Would not trigger:

```swift
@NSApplicationMain
final class AppDelegate: NSObject, NSApplicationDelegate {
	func applicationDidFinishLaunching(_ notification: Notification) {
		NSApp.activate(ignoringOtherApps: false)
	}
}
```

```swift
@NSApplicationMain
final class AppDelegate: NSObject, NSApplicationDelegate {
	func foo() {
		NSApp.activate(ignoringOtherApps: true)
	}
}
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

No.

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Enabled by default.
